service/javadsl/integration-client/src/main/java/com/lightbend/lagom/javadsl/client/integration/LagomClientFactory.java

```diff
@@ -243,11 +253,20 @@ private void closeGracefully(Effect close) {
     private static class StaticServiceLocator extends CircuitBreakingServiceLocator {
         private final URI uri;
 
-        StaticServiceLocator(CircuitBreakers circuitBreakers, URI uri) {
-            super(circuitBreakers);
+        StaticServiceLocator(CircuitBreakersPanel circuitBreakersPanel, URI uri) {
+            super(circuitBreakersPanel);
             this.uri = uri;
         }
 
+        /**
+         * @deprecated Use constructor accepting {@link CircuitBreakersPanel} instead
+         */
+        @Deprecated
+        StaticServiceLocator(com.lightbend.lagom.internal.client.CircuitBreakers circuitBreakers, URI uri) {
+            this(CircuitBreakersConverter.toJavadslCircuitBreakerInvoker(circuitBreakers), uri);
+        }
```

Comment:
I think it's that `LagomClientFactory` is just a bit weird. It is intended for use in non-Play, non-Lagom, non-Guice clients that need a simple way to build up the component structure.

In Scala, because there's no specific dependency injection framework, just constructor parameters and component traits, it is easier to reuse them in ordinary Scala code (although it might be nicer to have some convenience methods for creating them like in Java).

I don't think the service locator implementation classes should have been public in Scala, but I guess we're stuck with it now.